### ISSUE TITLE
Category and Label undefined

### ISSUE THREAD

**Issue by IHIutch** [sergeyfilimonov/gridsome-plugin-gtm#2]
Not sure if this is a bug with the plugin, vue-gtm, or my own poor understanding of GTM, but I was attempting to use the function below and kept getting "undefined" for category and label. Action was working fine.

I had set up Data Layer Variables specifically for each of the keys below with no luck. Any help would be greatly appreciated!

```
gtmTrackEvent(data) {
      this.$gtm.trackEvent({
        event: "mapPointClick",
        category: "Home",
        action: "Map Point Click",
        label: data.company,
        value: 1
      });
    },
```

**Comment by sergeyfilimonov** [comment]
@IHIutch hi! have you solved this issue for you?

**Comment by IHIutch** [comment]
No, I never was able to. I just ended up creating a Custom Event in Tag Manager and manually assigned the Category and Label there.

**Comment by pmcp** [comment]
I have the same  problem, @sergeyfilimonov ,  do you  have any idea what the  problem can be?

**Comment by sergeyfilimonov** [comment]
@pmcp @IHIutch I haven't actually used this feature. But have you checked **vue-gtm** documentation on this https://www.npmjs.com/package/vue-gtm#documentation? I'm not sure you can just call `gtmTrackEvent()` function without specifying the prototype. Have you tried `this.$gtm.trackEvent({})`?

**Comment by sergeyfilimonov** [comment]
@pmcp nope — I haven't used this feature myself in **vue-gtm**. Can you share your whole code which is not working and screenshot of error messages you get?

@pmcp @IHIutch also have you tried these solutions?

https://github.com/mib200/vue-gtm/issues/69
https://github.com/mib200/vue-gtm/issues/18

**Comment by edmeehan** [comment]
This might be the issue since GTM is a tag manager and not only for GA the inbound property name and outbound to dataLayer are different. Might have been on accident, or for another reason.

If you view the event and data in the dataLayer you can see that:
category = target
label = target-properties

Not the property names used by GA, but you can just make a GTM var and map it back to the proper property name.

Hope this helps.

**Comment by edmeehan** [comment]
Defined here - https://developer.aliyun.com/mirror/npm/package/vue-gtm/v/1.0.0
